### PR TITLE
Initial implementation of timescaledb metrics integration

### PR DESCRIPTION
**Pull request by c-kruse**
Closes https://github.com/sensu/catalog/issues/106

### CODE REVIEW

File: integrations/timescaledb/timescaledb-metrics/CHANGELOG.md [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+The format is based on [Keep a Changelog][changelog] and this project adheres
+to [Semantic Versioning][semver].
+
+## Unreleased
+
+- *n/a*
+
+## [0.1.0]
+
+- Initial add of `timescaledb-metrics` integration.
```
Comment (calebhailey) [inline integrations/timescaledb/timescaledb-metrics/CHANGELOG.md:14]:
```suggestion
## Unreleased

- Initial add of `timescaledb-metrics` integration.
```

Comment (calebhailey) [reply integrations/timescaledb/timescaledb-metrics/CHANGELOG.md:14]: We still haven't fully dialed in the release process yet, but we do know we're going to version these with a date-style numbers, e.g. `20220304.0.0`. For now let's init changelogs with "unreleased" comments.

File: integrations/timescaledb/timescaledb-metrics/README.md [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+There are no compatible dashboards for this integration.
+
+## Setup
+
+<!-- Sensu Integration setup instructions, including Sensu agent configuration and external component configuration -->
+<!-- EXAMPLE: what configuration (if any) is required in a third-party service to enable monitoring? -->
+
+1. Set up a [TimescaleDB][timescaledb-docs] database with access for your sensu backend.
+
+    See [sensu/sensu-timescaledb-handler][sensu/sensu-timescaledb-handler-github]  docs for sensu-specific schema.
```
Comment (calebhailey) [inline integrations/timescaledb/timescaledb-metrics/README.md:26]:
```suggestion
   See [sensu/sensu-timescaledb-handler][sensu/sensu-timescaledb-handler-github] docs for recommendations.
    
   **Example**: 
    
   ```sql
   CREATE EXTENSION IF NOT EXISTS timescaledb CASCADE;
   CREATE TABLE metrics (
     time    TIMESTAMPTZ         NOT NULL,
     name    TEXT                NOT NULL,
     value   DOUBLE PRECISION    NULL,
     source  TEXT                NOT NULL,
     tags    JSONB
   );
   ```
```

File: integrations/timescaledb/timescaledb-metrics/README.md [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+
+2. Add the timescaledb-metrics integration and provide it with the TimescaleDB DSN.
+
+   This integration requires the following [Sensu Secrets][secrets]:
+
+   - `timescaledb_dsn`
+
+   _NOTE: this integration creates one or more Sensu Secrets using the "env" provider. The corresponding environment variables need to be set on every sensu-backend in the Sensu deployment. To add the environment variables, please modify `/etc/default/sensu-backend` or `/etc/sysconfig/sensu-backend` and restart the sensu-backend service(s)._
```
Comment (calebhailey) [inline integrations/timescaledb/timescaledb-metrics/README.md:34]:
```suggestion
```

I think we can take this out since you've configured a prompt to collect the DSN 👍

File: integrations/timescaledb/timescaledb-metrics/sensu-integration.yaml [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+  tags:
+    - timescaledb
+  contributors:
+    - "@sensu"
+  prompts:
+    - type: section
+      title: Secrets Management
+    - type: markdown
+      body: |
+        The TimescaleDB Metrics integration requires a secret DSN for connecting to TimescaleDB.
```
Comment (calebhailey) [inline integrations/timescaledb/timescaledb-metrics/sensu-integration.yaml:23]:
```suggestion
        The TimescaleDB Metrics integration requires a [Connection URI] (sometimes referred to as a "Data Source Name", or DSN) for connecting to TimescaleDB.
        
        [Connection URI]: https://www.postgresql.org/docs/current/libpq-connect.html#id-1.7.3.8.3.6
```

I couldn't find a single reference to "DSN" in the Postgres docs anywhere. Weird!

**Review by calebhailey — APPROVED** [review]
This looks great! I made a few suggestions but otherwise this is good to go!

File: integrations/timescaledb/timescaledb-metrics/sensu-integration.yaml (hunk 4 above)
Comment (c-kruse) [reply integrations/timescaledb/timescaledb-metrics/sensu-integration.yaml:23]: Glad you pointed that out! Here's an issue https://github.com/sensu/sensu-timescaledb-handler/issues/9

tl;dr What [postgres docs](https://www.postgresql.org/docs/14/libpq-connect.html#LIBPQ-CONNSTRING) call Connection Strings take two forms, key/value and connection URIs. I (for reasons that elude me) have always associated "DSN "with the key/value style, and "Connection URI" with the URI style.
